libs/utils/trace_analysis.py

```diff
@@ -970,138 +1030,140 @@ def getClusterFrequencyResidency(self, cluster):
             return None
 
         freq_dfr = self.trace.df('cpu_frequency')
-        cluster_freqs = freq_dfr[freq_dfr['cpu'].isin(cluster)]
+        # We assume that every CPU in a cluster always run at the same
+        # frequency, i.e. the frequency is scaled per-cluster not per-CPU.
+        # Hence, we can limit the cluster frequencies data to a single CPU
+        #TODO: we could check that this is valid in the trace and return a
+        #      warning in case it is not
+        cluster_freqs = freq_dfr[freq_dfr.cpu == _cluster[0]]
```

Comment:
For consistency within this file, can we rename all the dataframe variables to be either just "df", when only one is in scope, or name_df when needed. For example this should be cluster_df.